src/libsyntax/parse/parser.rs

```diff
@@ -3248,16 +3256,24 @@ impl<'a> Parser<'a> {
             match self.token {
               // expr(...)
               token::OpenDelim(token::Paren) => {
-                let es = self.parse_unspanned_seq(
+                match self.parse_unspanned_seq(
                     &token::OpenDelim(token::Paren),
                     &token::CloseDelim(token::Paren),
                     SeqSep::trailing_allowed(token::Comma),
                     |p| Ok(p.parse_expr()?)
-                )?;
-                hi = self.prev_span;
-
-                let nd = self.mk_call(e, es);
-                e = self.mk_expr(lo.to(hi), nd, ThinVec::new());
+                ) {
+                    Ok(es) => {
+                        let nd = self.mk_call(e, es);
+                        hi = self.prev_span;
+                        e = self.mk_expr(lo.to(hi), nd, ThinVec::new());
+                    }
+                    Err(mut err) => { // recover from parse error in argument list
```

Comment:
And here?

(+ the arm has some duplication with the previous one I highlighted)

Reply:
Moved the logic to its own function. @Centril, @petrochenkov, can you check the new code?